models/keyword_test.go

```diff
@@ -94,15 +94,84 @@ var _ = Describe("Keyword", func() {
 	})
 
 	Describe("#GetKeywords", func() {
-		It("returns user keywords", func() {
-			user := FabricateUser("John", "john@example.com", "secret")
-			FabricateKeyword("Buy domain", false, &user)
-			FabricateKeyword("Buy bike", false, &user)
-			user2 := FabricateUser("David", "david@example.com", "secret")
-			FabricateKeyword("Buy car", false, &user2)
+		Context("given the search query string is empty", func() {
+			It("returns user keywords", func() {
+				user := FabricateUser("John", "john@example.com", "secret")
+				FabricateKeyword("Buy domain", false, &user)
+				FabricateKeyword("Buy bike", false, &user)
+				user2 := FabricateUser("David", "david@example.com", "secret")
+				FabricateKeyword("Buy car", false, &user2)
 
-			keywords := models.GetKeywords(&user)
-			Expect(keywords.Count()).To(BeNumerically("==", 2))
+				keywords := models.GetKeywords(&user, "")
+				Expect(keywords.Count()).To(BeNumerically("==", 2))
+			})
+		})
+
+		Context("given the search query string is domain", func() {
+			It("returns matched user keywords", func() {
+				var keywordIds []int64
+				userKeywords := []*models.Keyword{}
+				user := FabricateUser("John", "john@example.com", "secret")
+				buyDomainKeyword := FabricateKeyword("Buy domain", false, &user)
+				purchaseDomainKeyword := FabricateKeyword("Purchase domain", false, &user)
+				FabricateKeyword("Buy bike", false, &user)
+				user2 := FabricateUser("David", "david@example.com", "secret")
+				FabricateKeyword("Buy car", false, &user2)
+
+				_, err := models.GetKeywords(&user, "domain").All(&userKeywords)
+				if err != nil {
+					Fail("Getting keyword failed: " + err.Error())
+				}
+
+				for _, v := range userKeywords {
+					keywordIds = append(keywordIds, v.Id)
+				}
+
+				expectedKeywordIds := []int64{buyDomainKeyword.Id, purchaseDomainKeyword.Id}
+
+				Expect(keywordIds).To(Equal(expectedKeywordIds))
+			})
+		})
+
+		Context("given the search query string is purchase", func() {
```

Comment:
Why having the same test with `purchase` and `car` (below)? 🤔 These tests are identical. Instead, there should be one generic test when a keyword is valid.